tests/integration/kafka_test.go

```diff
@@ -0,0 +1,211 @@
+
+// Returns the standard output, or fails testing if the command returned an error
+func runIntegration(t *testing.T, config func([]string) []string) (string, string, error) {
+	t.Helper()
+
+	command := make([]string, 0)
+	command = append(command, *binPath)
+	command = config(command)
+
+	stdout, stderr, err := helpers.ExecInContainer(*container, command)
+
+	if stderr != "" {
+		log.Debug("Integration command Standard Error: ", stderr)
+	}
+
+	return stdout, stderr, err
+}
+
+func ensureBrokerClusterReady(tries int) {
+	config := sarama.NewConfig()
+	config.Version = sarama.V1_0_0_0
+	config.ClientID = "nri-kafka"
+
+	saramaBroker := sarama.NewBroker("localhost:" + KAFKA1_PORT)
+	err := saramaBroker.Open(config)
+	if err != nil {
+		tries += 1
+		if tries > BROKER_CONN_MAX_RETRIES {
+			log.Error("Failed opening connection: %s", err)
+			os.Exit(1)
+		}
+		time.Sleep(500 * time.Millisecond)
+		ensureBrokerClusterReady(tries)
+	}
+	connected, err := saramaBroker.Connected()
+	if err != nil {
+		tries += 1
+		if tries > BROKER_CONN_MAX_RETRIES {
+			log.Error("failed checking if connection opened successfully: %s", err)
+			os.Exit(1)
+		}
+		time.Sleep(500 * time.Millisecond)
+		ensureBrokerClusterReady(tries)
+	}
+	if !connected {
+		tries += 1
+		if tries > BROKER_CONN_MAX_RETRIES {
+			log.Error("Broker is not connected: %s", err)
+			os.Exit(1)
+		}
+		time.Sleep(500 * time.Millisecond)
+		ensureBrokerClusterReady(tries)
+	}
+
+	metadata, err := saramaBroker.GetMetadata(&sarama.MetadataRequest{})
+	if err != nil {
+		tries += 1
+		if tries > BROKER_CONN_MAX_RETRIES {
+			log.Error("failed to get metadata from broker: %s", err)
+			os.Exit(1)
+		}
+		time.Sleep(500 * time.Millisecond)
+		ensureBrokerClusterReady(tries)
+	}
+
+	if metadata == nil ||
+		metadata.Brokers == nil ||
+		(metadata != nil && metadata.Brokers != nil && len(metadata.Brokers) < BROKERS_IN_CLUSTER) {
+		tries += 1
+		if tries > BROKER_CONN_MAX_RETRIES {
+			log.Error("failed to start all brokers")
+			os.Exit(1)
+		}
+		time.Sleep(500 * time.Millisecond)
+		ensureBrokerClusterReady(tries)
+	}
+}
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+	ensureBrokerClusterReady(0)
+	result := m.Run()
+	os.Exit(result)
+}
+
+func zookeeperDiscoverConfig(command []string) []string {
+	return append(
+		command,
+		"--cluster_name", "kfk-cluster-zookeeper",
+		"--zookeeper_hosts", `[{"host": "zookeeper", "port": 2181}]`,
+		"--autodiscover_strategy", "zookeeper",
+		"--collect_broker_topic_data", "true",
+		"--topic_mode", "all",
+		"--topic_bucket", "3/3",
+	)
+}
+
+func bootstrapDiscoverConfig(command []string) []string {
+	return append(
+		command,
+		"--cluster_name", "kfk-cluster-bootstrap",
+		"--autodiscover_strategy", "bootstrap",
+		"--bootstrap_broker_host", "kafka1",
+		"--bootstrap_broker_kafka_port", "9092",
+		"--bootstrap_broker_kafka_protocol", "PLAINTEXT",
+		"--bootstrap_broker_jmx_port", "1099",
+		"--collect_broker_topic_data",
+		"--topic_mode", "all",
+	)
```

Comment:
For readability is often better to use `strings.Split("--cluster-name this --foo bar whatever", " ")`, as it makes easier to add commands/flags later

Reply:
For me it's better to see it in multiple lines.

I could agree to have every key value separated like that, so :

```
return append(		
               command,		
               strings.Split("--cluster_name kfk-cluster-bootstrap", " "),
               strings.Split("--autodiscover_strategy bootstrap"", " "),
```
               
But having a kilometric string with all the parameters obfuscates the visibility of the flags.

Reply:
What do you think?

Reply:
I didn't think about the linne being that long... Let's keep is as you have it for now :)